### PR TITLE
Copy ComponentResources struct to eliminate CRD confusion

### PR DESCRIPTION
**Pull request by vberezny**
## Description
[Bug Fix] Copy ComponentResources struct to make generated CRDs accurately reflect supported components and their resource requirements.

## For PR author

- [x ] Tests for change.
- [x ] If changing pkg/apis/, run `make gen-files`

## Issue
https://tigera.atlassian.net/browse/SAAS-1250

### CODE REVIEW

File: api/v1/logstorage_types.go [hunk 1]
```diff
@@ -142,6 +142,22 @@ type Retention struct {
...
+
+const (
+	ComponentNameECKOperator LogStorageComponentName = "ECKOperator"
+)
+
+// The ComponentResource struct associates a ResourceRequirements with a component by name
+type LogStorageComponentResource struct {
+	// ComponentName is an enum which identifies the component
+	// +kubebuilder:validation:Enum=ECKOperator
+	LogStorageComponentName LogStorageComponentName `json:"componentName"`
```
Comment (tmjd) [inline api/v1/logstorage_types.go:156]: I think we should just call this ComponentName, WDYT?
Either way this name and the comment above should match.

**Review by tmjd — COMMENTED** [review]
Looks good, one small naming thing to fix.

File: api/v1/logstorage_types.go (hunk 1 above)
Comment (vberezny) [reply api/v1/logstorage_types.go]: Yeah good call, changed it to `ComponentName`.

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**